devtools/server/actors/replay/connection.js

```diff
@@ -1144,7 +1156,7 @@ async function uploadAllSourcemapAssets({
     // once that is detected by the sources.
     sourceMapURL.startsWith("data:") ? undefined : ensureMapUploading(),
     Promise.all(sources.map(async ({ offset, url }) => {
-      const result = await fetchText(recordingId, url);
+      const result = await fetchText(browser.contentPrincipal, recordingId, url);
```

Comment:
Since `browser.contentPrincipal` here won't run until after we've fetch the sourcemap text, the principal here could be different from the one we passed on line 1125. We need to read `browser.contentPrincipal` just a single time and pass that principal to both calls to `fetchText`.

Reply:
Makes sense. Thanks!